src/main/java/io/cdap/plugin/gcp/bigquery/sink/AbstractBigQuerySink.java

```diff
@@ -112,6 +118,19 @@ public final void prepareRun(BatchSinkContext context) throws Exception {
 
   @Override
   public final void onRunFinish(boolean succeeded, BatchSinkContext context) {
+    if (succeeded) {
+      Job queryJob = bigQuery.getJob(jobId);
```

Comment:
I believe we've had bugs in the past related to jobs in different locations. Be sure to test writing to datasets in non-default locations to make sure this works. You may have to use the method that takes a JobId and set the location on that JobId.

Reply:
Thanks for pointing this out. Jobs were not being fetched for a different region. Fixed it.